.github/workflows/check.yml

```
@@ -1,16 +1,45 @@
-name: ESLint and Prettier
+name: CI Checks
 
 on:
   push:
-    branches: [ master ]
     paths:
+        # rerun when typescript files change
       - "ts/**"
-  pull_request:
-    branches: [ master ]
+        # rerun when workflow is changed
+      - ".github/workflows/check.yml"
+        # rerun when packages change
+      - "package.json"
+      - "package-lock.json"
+  pull_request: {}
```

```suggestion
  pull_request:
```

Is the empty dict needed?

---

Probably not actually! I'll try it without!

---

Yep seems to work fine!